spec/factories/provider.rb

```diff
@@ -2,7 +2,7 @@
   factory :provider do
     sequence(:name) { |n| "provider_#{seq_padded_for_sorting(n)}" }
     guid            { SecureRandom.uuid }
-    zone            { Zone.first || FactoryGirl.create(:zone) }
+    zone            { Zone.visible.first || FactoryGirl.create(:zone) }
```

Comment:
I don't think any of these `Zone.first` things should exist. (Unfortunately not really related to your PR intention)

Reply:
I don't know. And I'm pretty sure that no one is 100% sure for all specs. For example in EMS factory it was necessary for some completely unrelated specs.

Reply:
Same as above, this should be removed now that #18118 is merged.